### PR TITLE
add some const to mtype.d

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dmd/mtype.d [hunk 1]
```diff
@@ -6468,7 +6468,7 @@ extern (C++) final class TypeTuple : Type
             {
                 for (size_t i = 0; i < tt.arguments.dim; i++)
                 {
-                    Parameter arg1 = (*arguments)[i];
+                    const Parameter arg1 = (*arguments)[i];
                     Parameter arg2 = (*tt.arguments)[i];
```
Comment (jacob-carlborg) [inline src/dmd/mtype.d]: Can this not be made `const`?

**Review by RazvanN7 — APPROVED** [review]

File: src/dmd/mtype.d (hunk 1 above)
Comment (RazvanN7) [reply src/dmd/mtype.d]: why not?

Comment (jacob-carlborg) [reply src/dmd/mtype.d]: Because it's not marked as `const`?

Comment (RazvanN7) [reply src/dmd/mtype.d]: Oh sorry, I was under the impression you were asking that arg1 is not made `const`. As for `arg2`, I see no reason why it shouldn't

Comment (WalterBright) [reply src/dmd/mtype.d]: It can't be until the entire hierarchy of equals() marks it const.

Comment (jacob-carlborg) [reply src/dmd/mtype.d]: Ok.

**Pull request merged**